cmd/aro/deploy.go

```diff
@@ -38,6 +38,11 @@ func deploy(ctx context.Context, log *logrus.Entry) error {
 		return err
 	}
 
+	err = config.CheckRequiredFields()
```

Comment:
Follow-up: All this looks like an implementation detail of Deployer. 
Config is just config. 
Deployer cares about its validity. So it should be part of `New` of the deployment with something like `config.validate()`
At the same time `deployer` has logger in it so you can avoid `fmt.` statements. 

Please raise gh follow-up issue and work on this at some point.

Reply:
@petrkotas ^^